src/MainMenu.cpp

```diff
@@ -1637,4 +1637,58 @@ ChainCallbackResult MainMenu::OnDraw(MainMenu *menu)
 }
 #pragma optimize("", on)
 
+#pragma optimize("s", on)
+#pragma var_order(stageNum, color, charShotType, selectedStage, textPos, local_28, stage)
+ZunResult MainMenu::ChoosePracticeLevel()
+{
+    if (this->gameState == STATE_PRACTICE_LVL_SELECT)
+    {
+        i32 local_28;
+        D3DXVECTOR3 textPos(320.0, 200.0, 0.0);
+        if (this->stateTimer < 30)
+        {
+            local_28 = this->stateTimer * 0xFF / 30;
+        }
+        else
+        {
+            local_28 = 0xff;
+        }
+
+        i32 color = local_28;
+        i32 charShotType = (g_GameManager.character << 1) + g_GameManager.shotType;
+        i32 stage;
+        if (g_GameManager.clrd[charShotType].difficultyClearedWithoutRetries[g_GameManager.difficulty] > 7)
```

Comment:
```suggestion
        if (g_GameManager.clrd[charShotType].difficultyClearedWithoutRetries[g_GameManager.difficulty] > 6)
```

The cmp is wrong right now.